app/controllers/routes_controller.rb

```diff
@@ -28,12 +24,15 @@ def route
     @route ||= Route.find(params[:id])
   end
 
-  def negative_length?
-    route_params[:length].negative?
+  def validate_ended_route
```

Comment:
why changing this to a callback? does it make code easier to understand?

Reply:
Because if not, rubocop throws problems with ABCSize

Reply:
mmm we should find a way to do it without the callback

Reply:
let me think about it

Reply:
Do not use the before action callback in this case. Increase the number of ABCSize if you need to, i will think another approach later.